jaxtyping/_array_types.py

```diff
@@ -744,3 +747,22 @@ class _Cls(AbstractDtype):
 Shaped = _make_dtype(_any_dtype, "Shaped")
 
 Key = _make_dtype(_prng_key, "Key")
+
+
+def make_numpy_struct_dtype(dtype, name):
+    """This is a convenience function to create type annotation for numpy structured array
+    It does exact match on the name, order, and dtype of all its fields
+
+    !!! Example:
+
+        ```python
+        label_t = np.dtype([('first', np.uint8), ('second', np.int8)])
+        Label = make_numpy_struct_dtype(label_t, Label)
+        ```
+        after that, you can use it just like any AbstractDtype
+        ```python
+        Label[np.array, 'a b c']
+        ```
+    """
+    assert isinstance(dtype, np.dtype)
```

Comment:
Nits:
- Can you add type annotations?
- Can you add `**Arguments:**` and `**Returns:**` sections? (See `jaxtyped` for an example.)
- Can you switch this from an `assert` to a `ValueError`? (The former I reserve for internal "developer mistake" errors, not "user passed in the wrong input" errors.)

Other than this, this PR looks good to me! Thank you for contributing :)